src/main/kotlin/no/nav/syfo/melding/status/kafka/KafkaDialogmeldingStatusConsumer.kt

```diff
@@ -0,0 +1,93 @@
+package no.nav.syfo.melding.status.kafka
+
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.application.kafka.KafkaConsumerService
+import no.nav.syfo.melding.database.getMelding
+import no.nav.syfo.melding.status.MeldingStatusService
+import no.nav.syfo.melding.status.database.*
+import no.nav.syfo.melding.status.domain.MeldingStatusType
+import org.apache.kafka.clients.consumer.ConsumerRecords
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import java.time.Duration
+import java.util.UUID
+
+class KafkaDialogmeldingStatusConsumer(
+    private val database: DatabaseInterface,
+    private val meldingStatusService: MeldingStatusService,
+) : KafkaConsumerService<KafkaDialogmeldingStatusDTO> {
+    override val pollDurationInMillis: Long = 1000
+    override fun pollAndProcessRecords(kafkaConsumer: KafkaConsumer<String, KafkaDialogmeldingStatusDTO>) {
+        val records = kafkaConsumer.poll(Duration.ofMillis(pollDurationInMillis))
+        if (records.count() > 0) {
+            processRecords(
+                consumerRecords = records,
+            )
+            kafkaConsumer.commitSync()
+        }
+    }
+
+    private fun processRecords(consumerRecords: ConsumerRecords<String, KafkaDialogmeldingStatusDTO>) {
+        database.connection.use { connection ->
+            consumerRecords.forEach {
+                COUNT_KAFKA_CONSUMER_DIALOGMELDING_STATUS_READ.increment()
+                val kafkaDialogmeldingStatus = it.value()
+                if (kafkaDialogmeldingStatus != null) {
+                    processDialogmeldingStatus(
+                        kafkaDialogmeldingStatus = kafkaDialogmeldingStatus,
+                        connection = connection,
+                    )
+                } else {
+                    COUNT_KAFKA_CONSUMER_DIALOGMELDING_STATUS_TOMBSTONE.increment()
+                    log.warn("Received KafkaDialogmeldingStatusDTO with no value: could be tombstone")
+                }
+            }
+            connection.commit()
+        }
+    }
+
+    private fun processDialogmeldingStatus(
+        kafkaDialogmeldingStatus: KafkaDialogmeldingStatusDTO,
+        connection: Connection,
+    ) {
+        val meldingUuid = UUID.fromString(kafkaDialogmeldingStatus.bestillingUuid)
+        val meldingId = database.getMelding(uuid = meldingUuid)?.id
+        if (meldingId != null) {
+            log.info("Received KafkaDialogmeldingStatusDTO for known melding: meldingUuid $meldingUuid")
+            createOrUpdateMeldingStatus(
+                connection = connection,
+                meldingId = meldingId,
+                kafkaDialogmeldingStatus = kafkaDialogmeldingStatus,
+            )
+        } else {
+            COUNT_KAFKA_CONSUMER_DIALOGMELDING_STATUS_SKIPPED.increment()
+            log.info("Received KafkaDialogmeldingStatusDTO but no melding found: meldingUuid $meldingUuid")
```

Comment:
Litt usikker på om det er lurt å logge noe her - det vil være mange apprecs/meldingstatuser på topic som ikke har noe med meldinger fra isbehandlerdialog å gjøre, ikke sant?

Reply:
Ja, det vil jo komme statuser for alle meldinger som "bestilles" fra isdialogmelding så vi kan godt fjerne loggingen.